src/Tarjan.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Graphp\Algorithms;
+
+use Fhaculty\Graph\Exception\InvalidArgumentException;
+use Fhaculty\Graph\Graph;
+use Fhaculty\Graph\Set\Vertices;
+use Fhaculty\Graph\Vertex;
+
+/**
+ * @see http://github.com/Trismegiste/Mondrian/blob/master/Graph/Tarjan.php
+ */
+class Tarjan extends BaseGraph
+{
+    private $stack;
+    private $index;
+    private $partition;
+
+    private $indexMap = array();
+    private $lowLinkMap = array();
+
+    /**
+     * @param Graph $graph
+     */
+    public function __construct(Graph $graph)
+    {
+        parent::__construct($graph);
+        $this->indexMap = new \SplObjectStorage();
+        $this->lowLinkMap = new \SplObjectStorage();
+    }
+
+    /**
+     * Get the strongly connected components of this digraph by
+     * the Tarjan algorithm.
+     *
+     * Starting from :
+     * http://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm
+     *
+     * Corrected with the help :
+     * https://code.google.com/p/jbpt/source/browse/trunk/jbpt-core/src/main/java/org/jbpt/algo/graph/StronglyConnectedComponents.java
+     * @throws \Fhaculty\Graph\Exception\InvalidArgumentException
+     * @return array the partition of this graph : an array of an array of vertices
+     */
+    public function getStronglyConnected()
```

Comment:
The method signature should reflect what goes in what goes out. What exactly is this supposed to return? :)